lib/parsers/asm-parser-spirv.ts

```diff
@@ -105,13 +105,22 @@ export class SPIRVAsmParser extends AsmParser {
         const opLine = /OpLine/;
         const opNoLine = /OpNoLine/;
         const opExtDbg = /OpExtInst\s+%void\s+%\d+\s+Debug/;
+        const opExtDbgLine = /OpExtInst.*DebugLine %\w+\s+(%\w+)\s+(%\w+)/;
```

Comment:
some background here, SPIR-V had an `OpLine` way to add which line was used, but was very limited and a new set of [NonSemantic DebugInfo Instructions](https://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/main/nonsemantic/NonSemantic.Shader.DebugInfo.100.html) were added. The main issue is it involves a lot of indirect lookups. Since people are only testing single file and (currently) not combing many modules together, we can get away assuming a lot for sake of making this ASM Parsing not get too crazy

Reply:
yes! Everyone else still uses `OpLine` and will not use `OpExtInst ... DebugLine`, but if they do, it will now work (other tools like glslang and DXC that output SPIR-V can do both, but `slangc` just ditched adding the old way)

Reply:
I guess another way to put it, I have been testing this branch locally with GLSL/HLSL and they still work the same

Reply:
Thanks both